sbin/create_genesis.sh

```diff
@@ -25,8 +25,49 @@ relpath () {
 BASE_GENESIS_PATH=`relpath $BASE_GENESIS_PATH $CONFIG_DIR`
 GENESIS_PATH=`relpath $GENESIS_PATH $CONFIG_DIR`
 
+L1_WS_PORT=`echo $L1_ENDPOINT | awk -F':' '{print $3}'`
+echo "Parsed port: $L1_WS_PORT from $L1_ENDPOINT"
+
+# Start an L1 if one isn't already running
+if ! ss -tuln | grep -q ":$L1_WS_PORT "; then
+    echo "Starting L1..."
+    if [ $L1_STACK = "geth" ]; then
+        $GETH_BIN \
+        --dev \
+        --dev.period 1 \
+        --verbosity 3 \
+        --http \
+        --http.api eth,web3,net \
+        --http.addr 0.0.0.0 \
+        --ws \
+        --ws.api eth,net,web3 \
+        --ws.addr 0.0.0.0 \
+        --ws.port $L1_WS_PORT > /dev/null 2>&1 &
+        sleep 3
+        L1_PID=$!
+
+    elif [ $L1_STACK = "hardhat" ]; then
+        cd $CONTRACTS_DIR && npx hardhat node --no-deploy --port $L1_WS_PORT &
+        L1_PID=$!
+        echo "L1 PID: $L1_PID"
+        sleep 3
+    else
+        echo "invalid value for L1_STACK: $L1_STACK"
+        exit 1
+    fi
+else
+    L1_PID=0
+    echo "L1 already started..."
+fi
+
+
```

Comment:
I duplicated a lot of code from `start_l1.sh` but I didn't want to create an entire docker process just to fetch a single block